pkg/operator/ceph/object/controller.go

```diff
@@ -203,10 +207,16 @@ func (r *ReconcileCephObjectStore) reconcile(request reconcile.Request) (reconci
 	// DELETE: the CR was deleted
 	if !cephObjectStore.GetDeletionTimestamp().IsZero() {
 		logger.Debugf("deleting store %q", cephObjectStore.Name)
+
+		err := r.verifyObjectBucketCleanup(cephObjectStore)
+		if err != nil {
+			return reconcile.Result{}, errors.Wrapf(err, "failed to delete store %q", cephObjectStore.Name)
```

Comment:
The operator should continue attempting to delete the object store by re-queueing the event. So instead of returning an empty result, we need a re-queue event here.

Reply:
Returning error in the `Reconcile` would re-queue the event and would try to delete the object again.

Reply:
If the err is returned, you'll get immediate retries with exponential backoff. The reason I'd suggest a requeue event is so we can control the retry interval. In this case we know it can take time for the buckets to be deleted by the admin, so the exponential backoff doesn't make as much sense.

Reply:
Thanks for pointing that out and sharing a reference on how to do that. I've updated the code to add retry interval for reconcile.